lint/backend.py

```diff
@@ -90,13 +90,18 @@ def get_lint_tasks(linters, view, view_has_changed):
 
 
 def execute_lint_task(linter, code, offset, view_has_changed, settings, task_name):
-    # We 'name' our threads, for logging purposes.
-    threading.current_thread().name = task_name
+    try:
+        # We 'name' our threads, for logging purposes.
+        threading.current_thread().name = task_name
 
-    errors = linter.lint(code, view_has_changed, settings) or []
-    translate_lineno_and_column(errors, offset)
+        errors = linter.lint(code, view_has_changed, settings) or []
+        translate_lineno_and_column(errors, offset)
 
-    return errors
+        return errors
+    except BaseException:
+        # Log while multi-threaded to get a nicer log message
+        logger.exception('Linter crashed.\n\n')
+        raise
```

Comment:
Could also `return None` here and have `await_futures` check for `None` results (or filter out `None` results wherever that is used). That will stop the exception from bubbling through the entire thread and into the future when it has, in fact, already been "handled".

`BaseException` isn't needed here as neither KeyboardInterrupt nor SystemExit will ever be thrown inside an ST plugin context, but w/e.

Reply:
`w/e`?

Otherwise a bit too lazy here.

Reply:
w/e = whatever